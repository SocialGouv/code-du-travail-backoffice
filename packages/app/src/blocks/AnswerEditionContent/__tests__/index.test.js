import { render } from "@testing-library/react";
import React from "react";

import AnswerEditionContentBlock from "..";

// Polyfill "document.getSelection()""
// https://gist.github.com/yckart/6435861
Object.defineProperty(document, "getSelection", {
  value: () => {
    return document.selection && document.selection.createRange().text;
  },
});

describe("blocks/<AnswerEditionContent />", () => {
  const props = {
    defaultValue: "Hello World!",
    onChange: jest.fn(),
  };

  const 位 = render(<AnswerEditionContentBlock {...props} />);

  it("should match snapshot", () => {
    expect(位.container).toMatchSnapshot();
    expect(位.queryByText("A Tag")).not.toBeInTheDocument();
    expect(位.queryByText("Another Tag")).not.toBeInTheDocument();
  });
});
