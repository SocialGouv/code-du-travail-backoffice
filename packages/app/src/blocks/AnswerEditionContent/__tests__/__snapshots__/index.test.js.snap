// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Contrib] blocks/<AnswerEditionContent /> should match snapshot 1`] = `
.emotion-10 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.emotion-9 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.emotion-2 {
  color: #444444;
  font-size: 0.875rem;
  width: auto !important;
}

.emotion-0 {
  height: 0.75rem;
  margin: 2px 3px 0;
  vertical-align: -1px;
  width: 0.75rem;
}

<div>
  <div
    class="emotion-10 emotion-11"
  >
    <div
      class="emotion-8 emotion-9"
    >
      <div
        class="ql-toolbar ql-snow"
      >
        <span
          class="ql-formats"
        >
          <button
            class="ql-undo emotion-2 emotion-3"
            title="Bouton annulant la dernière modification effectuée"
            type="button"
          >
            <img
              class="emotion-0 emotion-1"
              src="/static/images/undo.svg"
            />
          </button>
          <button
            class="ql-redo emotion-2 emotion-3"
            title="Bouton réappliquant la dernière modification annulée"
            type="button"
          >
            <img
              class="emotion-0 emotion-1"
              src="/static/images/redo.svg"
            />
          </button>
        </span>
        <span
          class="ql-formats"
        >
          <button
            class="ql-bold"
            title="Bouton formattant le texte sélectionné en gras"
            type="button"
          >
            <svg
              viewBox="0 0 18 18"
            >
               
              <path
                class="ql-stroke"
                d="M5,4H9.5A2.5,2.5,0,0,1,12,6.5v0A2.5,2.5,0,0,1,9.5,9H5A0,0,0,0,1,5,9V4A0,0,0,0,1,5,4Z"
              />
               
              <path
                class="ql-stroke"
                d="M5,9h5.5A2.5,2.5,0,0,1,13,11.5v0A2.5,2.5,0,0,1,10.5,14H5a0,0,0,0,1,0,0V9A0,0,0,0,1,5,9Z"
              />
               
            </svg>
          </button>
          <button
            class="ql-italic"
            title="Bouton formattant le texte sélectionné en italique"
            type="button"
          >
            <svg
              viewBox="0 0 18 18"
            >
               
              <line
                class="ql-stroke"
                x1="7"
                x2="13"
                y1="4"
                y2="4"
              />
               
              <line
                class="ql-stroke"
                x1="5"
                x2="11"
                y1="14"
                y2="14"
              />
               
              <line
                class="ql-stroke"
                x1="8"
                x2="10"
                y1="14"
                y2="4"
              />
               
            </svg>
          </button>
          <button
            class="ql-underline"
            title="Bouton formattant le texte sélectionné en souligné"
            type="button"
          >
            <svg
              viewBox="0 0 18 18"
            >
               
              <path
                class="ql-stroke"
                d="M5,3V9a4.012,4.012,0,0,0,4,4H9a4.012,4.012,0,0,0,4-4V3"
              />
               
              <rect
                class="ql-fill"
                height="1"
                rx="0.5"
                ry="0.5"
                width="12"
                x="3"
                y="15"
              />
               
            </svg>
          </button>
        </span>
        <span
          class="ql-formats"
        >
          <button
            class="ql-link"
            title="Bouton créant un lien à partir du text sélectionné"
            type="button"
          >
            <svg
              viewBox="0 0 18 18"
            >
               
              <line
                class="ql-stroke"
                x1="7"
                x2="11"
                y1="7"
                y2="11"
              />
               
              <path
                class="ql-even ql-stroke"
                d="M8.9,4.577a3.476,3.476,0,0,1,.36,4.679A3.476,3.476,0,0,1,4.577,8.9C3.185,7.5,2.035,6.4,4.217,4.217S7.5,3.185,8.9,4.577Z"
              />
               
              <path
                class="ql-even ql-stroke"
                d="M13.423,9.1a3.476,3.476,0,0,0-4.679-.36,3.476,3.476,0,0,0,.36,4.679c1.392,1.392,2.5,2.542,4.679.36S14.815,10.5,13.423,9.1Z"
              />
               
            </svg>
          </button>
          <button
            class="ql-blockquote"
            title="Bouton formattant le paragraphe sélectionné au format citation"
            type="button"
          >
            <svg
              viewBox="0 0 18 18"
            >
               
              <rect
                class="ql-fill ql-stroke"
                height="3"
                width="3"
                x="4"
                y="5"
              />
               
              <rect
                class="ql-fill ql-stroke"
                height="3"
                width="3"
                x="11"
                y="5"
              />
               
              <path
                class="ql-even ql-fill ql-stroke"
                d="M7,8c0,4.031-3,5-3,5"
              />
               
              <path
                class="ql-even ql-fill ql-stroke"
                d="M14,8c0,4.031-3,5-3,5"
              />
               
            </svg>
          </button>
        </span>
        <span
          class="ql-formats"
        >
          <button
            class="ql-list"
            title="Bouton formattant le paragraphe sélectionné au format liste ordonnée"
            type="button"
            value="ordered"
          >
            <svg
              viewBox="0 0 18 18"
            >
               
              <line
                class="ql-stroke"
                x1="7"
                x2="15"
                y1="4"
                y2="4"
              />
               
              <line
                class="ql-stroke"
                x1="7"
                x2="15"
                y1="9"
                y2="9"
              />
               
              <line
                class="ql-stroke"
                x1="7"
                x2="15"
                y1="14"
                y2="14"
              />
               
              <line
                class="ql-stroke ql-thin"
                x1="2.5"
                x2="4.5"
                y1="5.5"
                y2="5.5"
              />
               
              <path
                class="ql-fill"
                d="M3.5,6A0.5,0.5,0,0,1,3,5.5V3.085l-0.276.138A0.5,0.5,0,0,1,2.053,3c-0.124-.247-0.023-0.324.224-0.447l1-.5A0.5,0.5,0,0,1,4,2.5v3A0.5,0.5,0,0,1,3.5,6Z"
              />
               
              <path
                class="ql-stroke ql-thin"
                d="M4.5,10.5h-2c0-.234,1.85-1.076,1.85-2.234A0.959,0.959,0,0,0,2.5,8.156"
              />
               
              <path
                class="ql-stroke ql-thin"
                d="M2.5,14.846a0.959,0.959,0,0,0,1.85-.109A0.7,0.7,0,0,0,3.75,14a0.688,0.688,0,0,0,.6-0.736,0.959,0.959,0,0,0-1.85-.109"
              />
               
            </svg>
          </button>
          <button
            class="ql-list"
            title="Bouton formattant le paragraphe sélectionné au format liste numérotée"
            type="button"
            value="bullet"
          >
            <svg
              viewBox="0 0 18 18"
            >
               
              <line
                class="ql-stroke"
                x1="6"
                x2="15"
                y1="4"
                y2="4"
              />
               
              <line
                class="ql-stroke"
                x1="6"
                x2="15"
                y1="9"
                y2="9"
              />
               
              <line
                class="ql-stroke"
                x1="6"
                x2="15"
                y1="14"
                y2="14"
              />
               
              <line
                class="ql-stroke"
                x1="3"
                x2="3"
                y1="4"
                y2="4"
              />
               
              <line
                class="ql-stroke"
                x1="3"
                x2="3"
                y1="9"
                y2="9"
              />
               
              <line
                class="ql-stroke"
                x1="3"
                x2="3"
                y1="14"
                y2="14"
              />
               
            </svg>
          </button>
        </span>
        <span
          class="ql-formats"
        >
          <span
            class="ql-header ql-picker"
          >
            <span
              aria-controls="ql-picker-options-0"
              aria-expanded="false"
              class="ql-picker-label"
              data-label="Normal"
              data-value=""
              role="button"
              tabindex="0"
            >
              <svg
                viewBox="0 0 18 18"
              >
                 
                <polygon
                  class="ql-stroke"
                  points="7 11 9 13 11 11 7 11"
                />
                 
                <polygon
                  class="ql-stroke"
                  points="7 7 9 5 11 7 7 7"
                />
                 
              </svg>
            </span>
            <span
              aria-hidden="true"
              class="ql-picker-options"
              id="ql-picker-options-0"
              tabindex="-1"
            >
              <span
                class="ql-picker-item"
                data-label="Normal"
                data-value=""
                role="button"
                tabindex="0"
              />
              <span
                class="ql-picker-item"
                data-label="Titre"
                data-value="2"
                role="button"
                tabindex="0"
              />
              <span
                class="ql-picker-item"
                data-label="Sous-titre"
                data-value="3"
                role="button"
                tabindex="0"
              />
            </span>
          </span>
          <select
            class="ql-header"
            style="display: none;"
          >
            <option
              selected=""
              value=""
            >
              Normal
            </option>
            <option
              value="2"
            >
              Titre
            </option>
            <option
              value="3"
            >
              Sous-titre
            </option>
          </select>
        </span>
      </div>
      <div
        class="ql-container ql-snow"
        width="1"
      >
        <div
          class="ql-editor"
          contenteditable="true"
          data-gramm="false"
        >
          <p>
            Hello World!
          </p>
        </div>
        <div
          class="ql-clipboard"
          contenteditable="true"
          tabindex="-1"
        />
        <div
          class="ql-tooltip ql-hidden"
        >
          <a
            class="ql-preview"
            href="about:blank"
            rel="noopener noreferrer"
            target="_blank"
          />
          <input
            data-formula="e=mc^2"
            data-link="https://quilljs.com"
            data-video="Embed URL"
            type="text"
          />
          <a
            class="ql-action"
          />
          <a
            class="ql-remove"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
